src/builder-git.c

```diff
@@ -868,3 +868,32 @@ builder_git_checkout (const char     *repo_location,
 
   return TRUE;
 }
+
+char *
+builder_git_get_default_branch (const char *repo_location)
+{
+  char *default_branch;
+  char ** parts;
+  char ** parts2;
+  g_autofree gchar *output = NULL;
+  g_autoptr(GError) error = NULL;
+
+  if (!git (NULL, &output, 0, &error,
+            "ls-remote", "--symref", repo_location, "HEAD", NULL))
+    return g_strdup ("master");
+
+  parts = g_strsplit (output, "\t", 1);
+  if (g_str_has_prefix (parts[1], "HEAD") && g_strstr_len (parts[0], -1, "ref: "))
+    {
+      parts2 = g_strsplit (parts[0], "/", -1);
```

Comment:
I feel like you can avoid this allocation entirely with `strrchr()` or similar?

Not actually sure the first `g_strsplit()` is really necessary either but not a huge deal.

Reply:
This might work:

```c
char *
builder_git_get_default_branch (const char *repo_location)
{
  g_auto(GStrv) parts = NULL;
  g_autofree gchar *output = NULL;
  g_autoptr(GError) error = NULL;

  if (!git (NULL, &output, 0, &error,
            "ls-remote", "--symref", repo_location, "HEAD", NULL))
    return g_strdup ("master");

  parts = g_strsplit (output, "\t", 1);
  if (g_strcmp0 (parts[1], "HEAD") == 0 && g_str_has_prefix (parts[0], "ref: "))
    {
      char *branch = strrchr (parts[0], '/');
      if (branch != NULL)
        return g_strdup (branch + 1);
    }

  return g_strdup ("master");
}
```

Reply:
Nice 👍🏻  the `strrchr` use, I pushed your version but without the `g_str_has_prefix (parts[0], "ref: ")` because `ref: ` is possible to not be at the start of the output when there is warning of a redirect, like example below:
```
$ git ls-remote --symref http://gitlab.gnome.org/GNOME/evince.git HEAD
warning: redirigiendo a https://gitlab.gnome.org/GNOME/evince.git/
ref: refs/heads/main	HEAD
7a5ceb841874bef4b91282eee1025b0335a21795	HEAD
```